compaction/compaction_manager.cc

```diff
@@ -260,18 +260,29 @@ class user_initiated_backlog_tracker final : public compaction_backlog_tracker::
     virtual void remove_sstable(sstables::shared_sstable sst)  override { }
 };
 
+compaction_manager::compaction_state& compaction_manager::get_compaction_state(table* t) {
+    try {
+        return _compaction_state.at(t);
+    } catch (std::out_of_range&) {
+        // Note: don't dereference t as it might not exist
+        throw std::out_of_range(format("Compaction state for table [{}] not found", fmt::ptr(t)));
+    }
+}
+
```

Comment:
Nice :-)

Reply:
perhaps a runtime_error instead? @bhalevy as this is about a compaction attempt after table was stopped.

Reply:
okay, makes sense.

Reply:
Done in v2